packages/utils/src/index.ts

```diff
@@ -10,8 +10,8 @@ export function getRegisteredStyles(
 
   classNames.split(' ').forEach(className => {
     if (registered[className] !== undefined) {
-      registeredStyles.push(`${registered[className]};`)
-    } else {
+      registeredStyles.push(registered[className] as string)
```

Comment:
`registered` contains serialized styles that are already terminated by a `;` by the serialization code. It avoids allocating a new string.

Reply:
The `;` should definitely be changing the hash of strings passing through here, so that's expected.

The `;` feels overly defensive to me and I don't like that it's also creating more strings for the same reasons explained in the other comment. It would be nice to ensure that what goes in `cache.registered` is strictly `;` terminated, then we can safely do the change here and avoid modifying the strings everytime they pass here.

I'll revert it for now though, might require more investigation.

Reply:
For more context, I frequently see strings like this one:

![image](https://github.com/user-attachments/assets/2fed0292-fa5d-4b5a-9897-f76b9e1b5d85)